tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -528,3 +559,214 @@ void test_mem_domain_remove_partitions(void)
 
 	k_thread_join(&mem_domain_6_tid, K_FOREVER);
 }
+
+/**
+ * @brief Test access memory domain APIs allowed to supervisor threads only
+ *
+ * @details
+ * - Create a test case to use a memory domain APIs k_mem_domain_init()
+ *   and k_mem_domain_add_partition()
+ * - Run that test in kernel mode -no errors should happen.
+ * - Run that test in user mode -Zephyr fatal error will happen (reason 0),
+ *   because user threads can't use memory domain APIs.
+ * - At the same time test that system support the definition of memory domains.
+ *
+ * @ingroup kernel_memprotect_tests
+ *
+ * @see k_mem_domain_init(), k_mem_domain_add_partition()
+ */
+void test_mem_domain_api_kernel_thread_only(void)
```

Comment:
I'm wondering how sanitycheck can detect if the test case fails？ I don't see any zassert in the code.

Reply:
Necessary to use flag `valid_fault = true;` or use like `set_fault_valid(true);`